internal/aws/proxy/server_test.go

```diff
@@ -129,6 +129,7 @@ func TestHandlerNilBodyIsOk(t *testing.T) {
 	handler := srv.(*http.Server).Handler.ServeHTTP
 	req := httptest.NewRequest(http.MethodPost,
 		"https://xray.us-west-2.amazonaws.com/GetSamplingRules", nil)
+	req.Header.Set("Content-Length", "0")
```

Comment:
did you want to place an assertion in the test checking the `Content-Length` header value is indeed changed? This way we can prevent regressions.

Reply:
I tried to figure out but I couldn't find a way to check if value has changed because I can only receive response in the test function

Reply:
I have a better way to test the director function itself. We can take it out and make a separate function. This way we can test it. Perhaps, another PR would make sense for that

Reply:
this change test doesn't accomplish anything. Should I remove it?

Reply:
> this change test doesn't accomplish anything. Should I remove it?

No.  Please write a test that does ensure your change does what it intends to do.

Reply:
I don't have a way to test that change, the only way i could figure out was take that function out. That change shouldn't belong here. or please suggest an idea to just test my change in this PR